### PR TITLE
Add KubeVirt client for migrations

### PR DESCRIPTION
**Pull request by rmohr**
Add a migration client around the low level rest client, which provides
an equal interface like the kubernetes core client provides for all its
types.

### CODE REVIEW

File: pkg/kubecli/vm.go [hunk 1]
```diff
@@ -35,14 +35,15 @@ import (
 
 type KubevirtClient interface {
 	VM(namespace string) VMInterface
```
Comment (davidvossel) [inline pkg/kubecli/vm.go]: having this interface defined in vm.go seems a little odd.

File: pkg/kubecli/vm.go [hunk 2]
```diff
@@ -35,14 +35,15 @@ import (
...
+	Migration(namespace string) MigrationInterface
 }
 
 type kubevirt struct {
 	restClient *rest.RESTClient
 }
 
 func (k *kubevirt) VM(namespace string) VMInterface {
-	return &vms{k.restClient, namespace}
+	return &vms{k.restClient, namespace, "vms"}
```
Comment (davidvossel) [inline pkg/kubecli/vm.go]: This isn't really related to the patch, but it seems like we should define the literal "vms" at some point.

File: pkg/kubecli/vm.go (hunk 1 above)
Comment (rmohr) [inline pkg/kubecli/vm.go]: Restructured the files a little bit, so that it makes more sense

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

File: pkg/kubecli/vm.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import (
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/rest"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+)
+
+func (k *kubevirt) VM(namespace string) VMInterface {
+	return &vms{k.restClient, namespace, "vms"}
```
Comment (davidvossel) [inline pkg/kubecli/vm.go]: This is just a nitpick and doesn't really matter. Same for migration object.
```&vms{restClient: k.restClient, namespace: namespace, resource: "vms}```

**Comment by davidvossel** [comment]
functional tests passed for me

**Pull request merged**